src/lib/encoder.ts

```diff
@@ -170,6 +178,10 @@ const valueTypeEncodingMap = {
     encode: (value: string) => abiCoder.encodeParameter('bytes', value),
     decode: (value: string) => abiCoder.decodeParameter('bytes', value),
   },
+  'bool[]': {
+    encode: (value: boolean) => abiCoder.encodeParameter('bool[]', value),
+    decode: (value: string) => abiCoder.decodeParameter('bool[]', value),
+  },
```

Comment:
You added `boolean` but I think you would need to also add it here with `boolean[]`

Reply:
@Hugoo good catch.
I have added the return type `boolean[]` and some more tests.